analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -68,7 +66,17 @@ private static ProgramState LearnBranchingConstraint<T>(ProgramState state, IBin
 
         private static SymbolicConstraint BinaryConstraint(BinaryOperatorKind kind, SymbolicValue left, SymbolicValue right)
         {
-            if (left.HasConstraint<BoolConstraint>() && right.HasConstraint<BoolConstraint>())
+            if (left is null && right is null)
+            {
+                return null;
+            }
+            else if (left is null || right is null)
+            {
+                return kind == BinaryOperatorKind.Or && (left ?? right).HasConstraint(BoolConstraint.True)
+                    ? BoolConstraint.True
+                    : null;
```

Comment:
The same is true for BinaryOperatorKind.And (with BoolConstraint.False).

Reply:
I'll do another PR